backend/src/test/java/com/bakdata/conquery/io/xodus/stores/SerializingStoreDumpTest.java

```diff
@@ -0,0 +1,216 @@
+	
+	// Test data
+	private final ConceptQuery cQuery = new ConceptQuery(new CQReusedQuery(new ManagedExecutionId(new DatasetId("testD"), UUID.randomUUID())));
+	private final User user = new User("username","userlabel");
+	
+	@BeforeEach
+	public void init() {
+		tmpDir = Files.createTempDir();
+		env = Environments.newInstance(tmpDir);
+		config = new StorageConfig();
+	}
+	
+	@AfterEach
+	public void destroy() throws IOException {
+		env.close();
+		FileUtils.deleteDirectory(tmpDir);
+	}
+	
+	private <KEY, VALUE> SerializingStore<KEY, VALUE> createSerializedStore(StorageConfig config, Environment environment, Validator validator, IStoreInfo storeId) {
+		return new SerializingStore<>(config, new XodusStore(environment, storeId), validator, storeId);
+	}
+	
+	/**
+	 * Tests if entries with corrupted values are dumped.
+	 */
+	@Test
+	public void testCorruptValueDump() throws JSONException, IOException {
+		// Set dump directory to this tests temp-dir
+		config.setUnreadbleDataDumpDirectory(Optional.of(tmpDir));
+		
+		// Open a store and insert a valid key-value pair (UserId & User)
+		try (SerializingStore<UserId, User> store = createSerializedStore(config, env, Validators.newValidator(), StoreInfo.AUTH_USER)){
+			store.add(user.getId(), user);
+		}
+		
+		// Open that store again, with a different config to insert a corrupt entry (UserId & ManagedQuery)		
+		try (SerializingStore<UserId, QueryDescription> store = createSerializedStore(config, env, Validators.newValidator(), new CorruptableStoreInfo(StoreInfo.AUTH_USER.getXodusName(), UserId.class, QueryDescription.class))){
+			store.add(new UserId("testU2"), cQuery);
+		}
+		
+		// Reopen the store with the initial value and try to iterate over all entries (this triggers the dump or removal of invalid entries)
+		try (SerializingStore<UserId, User> store = createSerializedStore(config, env, Validators.newValidator(), StoreInfo.AUTH_USER)){
+			IterationStatistic expectedResult = new IterationStatistic();
+			expectedResult.setTotalProcessed(2);
+			expectedResult.setFailedKeys(0);
+			expectedResult.setFailedValues(1);
+			
+			// Iterate (do nothing with the entries themselves)
+			IterationStatistic result = store.forEach((k,v,s) -> {});
+			assertThat(result).isEqualTo(expectedResult);
+		}
+		
+		// Test if the correct number of dumpfiles was generated
+		Condition<File> dumpFileCond = new Condition<>(f -> f.getName().endsWith(SerializingStore.DUMP_FILE_EXTENTION) , "dump file");
+		assertThat(tmpDir.listFiles()).areExactly(1, dumpFileCond);
+		
+		// Test if the dump is correct
+		File dumpFile = getDumpFile(dumpFileCond);
+
+		assertThat((QueryDescription) Jackson.MAPPER.readerFor(QueryDescription.class).readValue(dumpFile)).isEqualTo(cQuery);
+	}
+
+	private File getDumpFile(Condition<File> dumpFileCond) {
+		File dumpFile = tmpDir.listFiles(new FileFilter() {
+
+			@Override
+			public boolean accept(File pathname) {
+				return dumpFileCond.matches(pathname);
```

Comment:
Das nimmt auch ein Lambda wie in PreprocessorCommand: `listFiles(((dir, name) -> name.endsWith(ConqueryConstants.EXTENSION_DESCRIPTION)));`